Hm, the failure seems relevant:

```--- FAIL: TestExampleObjectSchemas (0.25s)
	examples_test.go:353: ../examples/glusterfs/glusterfs-service.json: glusterfs-service does not have a test case defined
	examples_test.go:358: skipping : ../examples/openshift-origin/secret.json/secret
	examples_test.go:358: skipping : ../examples/elasticsearch/service-account.yaml/service-account
	examples_test.go:358: skipping : ../docs/user-guide/multi-pod.yaml/multi-pod
```

--- a/examples/examples_test.go
+++ b/examples/examples_test.go
@@ -194,6 +194,7 @@ func TestExampleObjectSchemas(t *testing.T) {
 		"../examples/glusterfs": {
 			"glusterfs-pod":       &api.Pod{},
 			"glusterfs-endpoints": &api.Endpoints{},
+			"glusterfs-service":   &api.Service{},
 		},
 		"../docs/user-guide/liveness": {
 			"exec-liveness": &api.Pod{},

--- a/examples/glusterfs/README.md
+++ b/examples/glusterfs/README.md
@@ -75,6 +75,15 @@ NAME                ENDPOINTS
 glusterfs-cluster   10.240.106.152:1,10.240.79.157:1
 ```
 
+We need also create a service for this endpoints, so that the endpoints will be persistented. We will add this service without a selector to tell Kubernetes we want to add its endpoints manually. You can see [glusterfs-service.json](glusterfs-service.json) for details.
+
+Use this command to create the service:
+
+```sh
+$ kubectl create -f examples/glusterfs/glusterfs-service.json
+```
+
+
 ### Create a POD
 
 The following *volume* spec in [glusterfs-pod.json](glusterfs-pod.json) illustrates a sample configuration.

--- a/examples/glusterfs/glusterfs-service.json
+++ b/examples/glusterfs/glusterfs-service.json
@@ -1,0 +1,12 @@
+{
+  "kind": "Service",
+  "apiVersion": "v1",
+  "metadata": {
+    "name": "glusterfs-cluster"
+  },
+  "spec": {
+    "ports": [
+      {"port": 1}
+    ]
+  }
+}

